pages/c/_cluster/nodes/index.vue

```diff
@@ -0,0 +1,41 @@
+<script>
```

Comment:
Is threre a reason this needs a custom page (or virtual type)?  It doesn't seem to do anything out of the ordinary.

To hide the create/from yaml button?  If so ignore that problem for now and I'll add a way to declare that the type is not creatable.  It also needs to look at RBAC and see if the schema says it's creatable, but that won't help here because nodes are even though there's no good reason to ever do it AFAIK.

For context, right now so I'm trying to eliminate custom pages like these, e.g. `pages/workload/*`.  They're essentially the same as the generic `c/_cluster/_resource/*` routes, but with maybe a different asyncData, and maybe rendering a `list/_resource.vue` component instead of the standard one.  

So I'm consolidating those because there's really no need to have a duplicate tree of routes and multiple places printing the header (e.g. oops, this type now doesn't support starring because it's not in this copy of the `<h1>`).

Separate pages make sense when it's really a different king of page (like Gatekeeper), but for basic CRUD we'll just add whatever customization is needed to the `{list,edit,detail}/_resource.vue` support.

(there's also too many create/edit/view modes so I'm consolidating those too, and too many labels & annotations viewers and editors).

Reply:
I'll go ahead and remove it. 

However, I'd like to say it might be worth acknowledging that the reason we're ending up with workload, node and gatekeeper pages is because the one size fits all solution doesn't appear to hold as well as we might like.

I think the Favorite case is actually a better argument for having a Heading component that everyone uses rather than a case for using the _resource/index. I remember raising concern over treating gatekeeper constraints as a special case (because it was obvious dashboard wanted to treat all resources the same way) but we wanted to do that anyway which means we're not going to be able to use the _resource/index for that page and I bet more will follow. Well it just so happens that the original implementation of Favorite didn't include adding Favorite to the constraint page. If instead we had a Heading component that all pages we're expected to use Favorite could've just been added to Heading.

I looked into adding a condition to the Create button in _resource/index to hide it in cases like this but couldn't find a good way to apply it globally and I'm very much against adding exceptional code that targets a specific resource to the generic implementation. And overriding the list page doesn't help because the buttons aren't something the override currently has access to.

Personally I'd really like to see the following:
- Overriding a page allows you to override the entire body
- Composable components are used to build the generic resource pages that can then be reused by the overrides.
- Refactor the generic page so that the majority is implemented as a component with slots. The page would be responsible for fetching the data and then either rendering the generic component or the override. Overrides will then be able to reuse the generic component and override the sections that are deemed necessary (This is effectively inversion of control).
- The override can define hooks to specify fetching more async data which the page then passes to the override.

Reply:
There _aren't_ (and shouldn't be) that many special cases which are part of the "stuff inside a cluster" route tree.  I think if you look at what's merged from my last PR you already have most of what you're asking for in some form or another, but somewhat flipped around (instead of a component with replaceable slots, a component which loads different implementations from `/{list,edit,detail}`).

Workloads was implemented as special but isn't really (to be fair I don't think custom list views existed at the time). So is no longer special with my last PR.  It's completely handled by custom list/detail/edit components, which (now) take over the entire body of the resource page (below the header) and can provide their own asyncData.

(Unrelated, there's a new `fetch() in 2.12 that we should switch to to replace much of the differences and confusion between using `data`, `asyncData` and `loadDeps` in one component)

Gatekeeper is the only thing left in `pages/c/_cluster/` now.  It's the first example of a 2nd pattern we'll have more of, a top-level which manages the lifecycle of some catalog app, and then child resources that are managed by that app hoisted up as children instead of showing them in the regular tree.  Istio, Monitoring, Logging, Pipeline, etc will all follow the same pattern.  Everything else is create/edit/view/list-ing resources, and handled by `ResourceDetail` and `/pages/c/_cluster/_resource/...`.

And Gatekeeper probably doesn't even actually need custom routes..  Some of this may require small changes to framework but:

-  The index page for managing the chart _could_ actually be done in `/list/gatekeeper.vue` instead.  (Though since it's not a _list_ at all, I think I still like it as a separate thing in the `/pages`)

- The list pages for constraint and template could AFAICT be regular `/list` components.  Workload does a similar thing of combining several resources into one rendered list.

- `moveType` should be able to move constraints and templates underneath the virtual OPA Gatekeeper entry in the tree..
  - Them not being virtual would also give you the count of templates and fix the highlight on the nav item when looking at the detail/edit of an instance.
  - For constraints you'd need a hook to provide a custom count implementation (which I think I did once before for workloads, but removed since they aren't countable from just counts data without loading all the resources)

- The constraint 'detail' and 'shared' are redundant since the default behavior is to use the `edit` component for viewing if there is no detail.